tmt/steps/discover/fmf.py

```diff
@@ -13,83 +13,119 @@
 
     discover:
         how: fmf
-        repository: https://github.com/psss/tmt
-        revision: master
-        destination: tmt
+        url: https://github.com/psss/tmt
+        ref: master
+        path: /fmf/root
+        test: /tests/basic
         filter: 'tier: 1'
 """
 
 import os
-import re
 import fmf
 import tmt
 import shutil
+import click
 import tmt.steps.discover
 
-from click import echo
-from fmf.utils import listed
-
 class DiscoverFmf(tmt.steps.discover.DiscoverPlugin):
-    """ Discover available tests from FMF metadata """
-
-    def __init__(self, data, step):
-        """ Check supported attributes """
-        super(DiscoverFmf, self).__init__(
-            data=data, step=step, name=data['name'])
-        self.tree = None
-        # Convert data into attributes for easy handling
-        self.repository = data.get('repository')
-        self.revision = data.get('revision')
-        filtr = data.get('filter', [])
-        self.filters = filtr if isinstance(filtr, list) else [filtr]
-        self._tests = []
+    """ Discover available tests from fmf metadata """
+
+    # Supported methods
+    _methods = [
+        tmt.steps.Method(
+            name='fmf',
+            summary='Flexible Metadata Format',
+            order=50),
+        ]
+
+    @classmethod
+    def options(cls, how=None):
+        """ Prepare command line options for given method """
+        return [
+            click.option(
+                '-u', '--url', metavar='REPOSITORY',
+                help='URL of the git repository with fmf metadata.'),
+            click.option(
+                '-r', '--ref', metavar='REVISION',
+                help='Branch, tag or commit specifying the git revision.'),
+            click.option(
+                '-p', '--path', metavar='ROOT',
+                help='Path to the metadata tree root.'),
+            click.option(
+                '-t', '--test', metavar='NAMES', multiple=True,
+                help='Select tests by name.'),
+            click.option(
+                '-F', '--filter', metavar='FILTERS', multiple=True,
+                help='Include only tests matching the filter.'),
+            ] + super().options(how)
+
+    def default(self, option, default=None):
+        """ Return default data for given option """
+        # Git revision defaults to master
+        if option == 'ref':
+            return 'master'
+        # No other defaults available
+        return default
+
+    def wake(self):
+        """ Wake up the plugin (override data with command line) """
+
+        # Handle backward-compatible stuff
```

Comment:
why we need to handle here some backward compatibility?

Reply:
The old examples used `repository` and `revision` instead of `url` and `ref` which are consistent with fmf identifier. Let's don't brake this for early adopters who have already used the old syntax.

Reply:
ok, I guess we can get rid of this after some time